JDXConverter.py

```diff
@@ -557,7 +600,118 @@ def startCommandLine(dataBaseFileName='MoleculesInfo.csv'):
+            SourceOfFragmentationPattern = molecule_meta_data_from_database[6]
+            SourceOfIonizationDatum = molecule_meta_data_from_database[7]
+
+        #Now we will check if the corresponding JDX file for the molecule exists in the local directory or not
+        elif(checkInLocalJDXDirectory(defaultJDXFilesLocation, moleculeName)):
+            #Now we will retrieve the spectrum information from the local JDX file
+            individual_spectrum.extend(getSpectrumDataFromLocalJDX([moleculeName]))
+            
+            #As the metadata for the molecule is not present inside the database csv file, we will now retrieve them from online
+            spectrum_data,molecular_formula,molecular_weight,electron_number,knownMoleculeIonizationTypeOnline, knownIonizationFactorRelativeToN2Online, SourceOfFragmentationPatternOnline, SourceOfIonizationDatumOnline = getMetaDataForMoleculeFromOnline(molecule_meta_data_from_database,moleculeName)
+            ENumber = int(electron_number)
+            MWeight = float(molecular_weight)
+            knownMoleculeIonizationType = knownMoleculeIonizationTypeOnline
+            knownIonizationFactorRelativeToN2 = knownIonizationFactorRelativeToN2Online
+            SourceOfFragmentationPattern = SourceOfFragmentationPatternOnline
+            SourceOfIonizationDatum = SourceOfIonizationDatumOnline
+
+        #Otherwise we will get all the metadata + spectrum data from online
+        else:
+            spectrum_data,molecular_formula,molecular_weight,electron_number,knownMoleculeIonizationTypeOnline, knownIonizationFactorRelativeToN2Online, SourceOfFragmentationPatternOnline, SourceOfIonizationDatumOnline = getMetaDataForMoleculeFromOnline(molecule_meta_data_from_database,moleculeName)
+            ENumber = int(electron_number)
+            MWeight = float(molecular_weight)
+            knownMoleculeIonizationType = knownMoleculeIonizationTypeOnline
+            knownIonizationFactorRelativeToN2 = knownIonizationFactorRelativeToN2Online
+            SourceOfFragmentationPattern = SourceOfFragmentationPatternOnline
+            SourceOfIonizationDatum = SourceOfIonizationDatumOnline
+
+            individual_spectrum.extend(spectrum_data)
+        
+        #Now we will add all the spectrum data and metadata into list like variables which will be passed into the exportToCSV function
+        ENumbers.append(ENumber)
+        MWeights.append(MWeight)
+        listOfJDXFileNames.append(JDXfilename)
+        knownMoleculeIonizationTypes.append(knownMoleculeIonizationType)
+        knownIonizationFactorsRelativeToN2.append(knownIonizationFactorRelativeToN2)
+        SourcesOfFragmentationPattern.append(SourceOfFragmentationPattern)
+        SourceOfIonizationData.append(SourceOfIonizationDatum)
+
+        AllSpectra = combineArray(AllSpectra,individual_spectrum)
+
+        print("ENTER a MOLECULE NAME or type END to stop entering molecule name")
+

```

Comment:
@XLR8-07 This function or block of code has what I would call "implied returns". When something is a global variable it is an "implied argument" or "implied return".  You should clarify here that these are implied returns. Ther is no exportToCSV function call here, but these lists are all global variables that will be used during exportToCSV.  (At least, that is the impression I have). You should write what the logic is in the comments here. I if have the logic wrong, then add the correct logic in the comments.

Reply:
Yes, your logic is correct, @AdityaSavara . These variables will be used for exportToCSV() function call. Regardless, I didnt understand the idea of "implied returns". Keen to know about it a bit more.

Reply:
The way to think about implied returns and implied arguments is like this:

A function normally has explicit arguments and explicit returns. Anyone looking at the code can see what the arguments and returns are.

When there are global variables, sometimes these must be populated before the function is run. That is an "implied" argument (because it is not an explicit argument).  Similarly, sometimes the information is extracted from the global variable at the end. This is an "implied" return (not an explicit return).  I use the word  "implied" but most people use the word "implicit".

Reply:
Clearly understood the concept, thanks for explaining.

Reply:
For the "implied arguments" and "implied returns" , you should always make a comment (at the top/bottom of function or code block) explaining what are the implied argument or implied return, that way a person does not get confused about such "hidden" logic.